### DISCUSSION TITLE
Will there eventually be a Web UI for dynamic rules?

### DISCUSSION THREAD

**Discussion by morphinestyle** [shlinkio/shlink#2044] [Help wanted] [answered]
Dynamic rules are great, but a bit tricky :D
Is there any plan for a Web UI management for setting up dynamic rules in the future?

**Comment by acelaya** [comment, marked as answer]
Yes, of course 🙂, it's mentioned in the [upgrade](https://github.com/shlinkio/shlink/blob/0bc741243042772b1522a791c6520ecc83e4e319/UPGRADE.md#from-v3x-to-v4x) doc:

> Support for dynamic rule-based redirects will be added to shlink-web-client soon, in v4.1.0

You can watch https://github.com/shlinkio/shlink-web-component/issues/271 and https://github.com/shlinkio/shlink-web-client/issues/1079

**Reply by acelaya** [reply]
This is a small quick pick

https://github.com/shlinkio/shlink/assets/2719332/8807cb38-f706-4bee-bcd6-987f73ed0184

**Reply by morphinestyle** [reply]
It's perfect, and above all very clear!

The ideal would be to be able to have a REDIRECTS column on the main links page which shows the types of redirects active on each link with small colored badges.

This would help a lot, over time, to identify whether a redirect has already been activated for a specific link: a small badge would help to avoid infinite verification "loops" for those who have many links.

I can already imagine myself constantly checking the links over time, doubting that I have forgotten to activate the redirects!  :D

**Reply by acelaya** [reply]
Yep, the plan is to add something like that. I also would like to allow ordering the rules via drag'n'drop when in a desktop device, but those improvements will likely come later.
